source/base/function_spherical.cc

```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```

Comment:
Hm, you are correct -- these arguments indeed do not exist... Must be the altitude here that made me think that they do...